src/webview/pipeline/app/common/resource.ts

```diff
@@ -0,0 +1,76 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { PipelineStart } from './types';
+
+
+export function parameter(paramName: string, defaultValue: string, initialValue: PipelineStart): void {
+  if (initialValue.params.length === 0) {
+    initialValue.params.push({name: paramName, default: defaultValue});
+  } else {
+    const found = initialValue.params.some(value => {
+      if (value.name === paramName) {
+        value.default = defaultValue;
+        return true;
+      }
+    });
+    if (!found) {
+      initialValue.params.push({name: paramName, default: defaultValue});
+    }
+  }
+}
+
+export function createResourceJson(resourceName: string, resourceReference: string, initialValue: PipelineStart): void {
```

Comment:
Same name problem, function named `create*` but return `void`

Reply:
I am storing the resource data here I will change the function name accordingly.